### PR TITLE
airbyte-ci: add dagger run url

### LINKED ISSUE #27863: connectors-ci: The Dagger UI url is not displayed in logs
All pipeline using dagger in our CI generate a Dagger Cloud run url that is helpful to debug pipeline execution and check what's cached or not.

This URL is only shared in the dagger logs that we persist at the end of the pipeline run to GCS.
It'd be great to get this url in the HTML report and in the GHA logs because it's helpful to track what's going on during pipeline execution.

We have two options to retrieve this url:
* Hacky: Read the dagger logs at runtime in airbyte-ci and parse it when we find the log line "Dagger Cloud run url"
* Ask the Dagger team if we can retrieve this url in a more elegant way with their SDK

### PR DESCRIPTION
**Pull request by pedroslopez**
## What

Adds a link to the Dagger Cloud runs for the given commit when running in CI. 

This is added to the reports as well as printed out to the console.

close https://github.com/airbytehq/airbyte/issues/27863

## How

Since we don't have a direct run ID, link to the Dagger Cloud changesByPipeline page, which lists all runs for a given commit.

Unfortunately this does mean that while the commit is the latest one on master, if anyone runs a publish or pre-release on master it will all get accumulated under that same page.

## Examples

- PR Comment: https://github.com/airbytehq/airbyte/pull/28947#issuecomment-1661208166
- [GHA Logs](https://github.com/airbytehq/airbyte/actions/runs/5732618701/job/15535896207?pr=28947#step:9:215)
- [Report](https://ffb3f9d8b3423fda74aa86e50a35cc5489ab98f5087ccbd7d269f4c-apidata.googleusercontent.com/download/storage/v1/b/airbyte-ci-reports-multi/o/airbyte-ci-internal%2Fconnectors%2Ftest%2Fpull_request%2Fpedroslopez_dagger-cloud-run-url%2F1690930340%2Fe54b0909115f3f9d00d4789324cb10ae11c8bbe6%2Fsource-faker%2F4.0.0%2Foutput.html?jk=ATmLuOG-Cd5Lg23uMg6WCvAAQfbaKJa2QvM2OPLFQ-LdbTRWOgyLLYoYbxAZ-93CIHnikituPBiKqf-8BpmJkcgC8lLooSfCf_xsRtPWL0VQ_Dyho1kTJwvHFdpE4MYgSeAFysJg801uu0b_7jMkn01VDm36792iVEAuLSnaggcPNCcnfhWBIdT_nVG_or3uyFSacvEPF11f5mzLGbJzC1OD9kgerRJT7Z6EwCTNE0KIYRfPslzgBSoP5RlhxBF3Sblq92yO2Db4is8SAsscuJVeXvC2mn9UljblA5WqOggxWDR7MjVfZqO6mDnweiu2TFh-qzN4dT-9iNYo0Q7wvkVPEJw93_aVvso9ABbkjimwN86pXYMzTRgpDNq1_UvcDs_3rmDjYPkjA2RhQVmFMvHR9agD6CqNg7AxEzGzAZvzWj-hgdC0lwHO34cppGDpWhSw8jrzfODaewe8LmWzM4B7Y6ALexhk75Y2uIcEeMXH8vDB5FUtoReimHZ5JCkkqy_G9fzqtind_1IOr_eanPO7dii8yjtKhXU8zo2EWj59HicP49HOGMC9QpSlQJTGpdvxAz1v47MGe_8biHGOMrGzv9qDKtIUYYXEroijMkjiOUsEwENnfGolUqrLEfbHMSAQg_BAJafJncwAW_haTe--Dr-nNDScT9TZiyWp0tvRasDRSyzKfIzno8pV39LVnYi-CorgtvF_QJgBdl48HLs5oaYML7bUHkg4iD6KJLycvOFqTEy0c1MN1Da5yDQ9o74E3OjbrYfe0B8Gv_eylVp_ezT-ypOBcvvhw-D2EKiLcBHoRUflSZGKjuJlAfLz8-7_7LQPSiajIHlk1a-Zfb3cgld5xFN7WqR6-WigL52bgEG-lwcUvUxBaFXIk_h44uU53UOw4TtY9sRUUfOjUXSdUh_dFM235CdWYBneoubna5uYSfqqYl08HHoqPtv2ksdz7Z6y6srAvmvN8XGmXz7ucJB9VKl1pRzDmwVk3Tli7ygyuiOyfMd8tAyvrTsPsnFim6zNWKfLvaG_HRaYbSwMedEVBCFQCTYPBu-y7tyCgoAeyc6h_UE38je9FSaYrhgJPMwsA9gTr3ifbtqkl4ABvGWyV1yhWz9P3KNvPh11TWZ03MTnN67HAZnQw4U3RKoDbOKTV4idUrzbCoS5kll9ybO3ZybAR55hWd3QvI7QulEhMFvFyfHMHJUxE1nOHbvXjh-uj7RDs_c1Xh4UOqyVTzGHQPL7eMFbap3G4v-DgEByIj5K7U_bjAB-RNrcMMH9mSd8x95wzOg0gpoOcx4xpkLswYVqxEM5YUBCVtuCJiK4GhmJxPqYqlmnoYGKeCTybO92jnsN64ysR_NOcx1kKM_FpJF-bmaVktj_h_HWIpnzoq6ojMDcp6ghR7aJPKMMx_WuUdnFsflcXUWHm-TCfGovsySyiK4lBRd_Mf0ppWVxhw&isca=1)

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`cbd3cdfe39`](https://github.com/airbytehq/airbyte/pull/28947/commits/cbd3cdfe397cfa963e92e6538e568289ac0688e2)) - ❌

⏲️  Total pipeline duration: 01mn53s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ❌        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci-internal/connectors/test/pull_request/pedroslopez_dagger-cloud-run-url/1690926524/cbd3cdfe397cfa963e92e6538e568289ac0688e2/source-faker/4.0.0/output.html)

☁️ [View commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:cbd3cdfe397cfa963e92e6538e568289ac0688e2)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`9feaba9f0b`](https://github.com/airbytehq/airbyte/pull/28947/commits/9feaba9f0b464fb8fd297937fb15ba3364018cb6)) - ❌

⏲️  Total pipeline duration: 02mn12s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ❌        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci-internal/connectors/test/pull_request/pedroslopez_dagger-cloud-run-url/1690927098/9feaba9f0b464fb8fd297937fb15ba3364018cb6/source-faker/4.0.0/output.html)

☁️ [View commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:9feaba9f0b464fb8fd297937fb15ba3364018cb6)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`cf08ebb8ee`](https://github.com/airbytehq/airbyte/pull/28947/commits/cf08ebb8ee4b5975f94a820ef080536abccd267e)) - ❌

⏲️  Total pipeline duration: 01mn46s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ❌        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci-internal/connectors/test/pull_request/pedroslopez_dagger-cloud-run-url/1690928425/cf08ebb8ee4b5975f94a820ef080536abccd267e/source-faker/4.0.0/output.html)

☁️ [View commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:cf08ebb8ee4b5975f94a820ef080536abccd267e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

File: airbyte-ci/connectors/pipelines/pipelines/tests/templates/test_report.html.j2 [hunk 1]
```diff
@@ -169,6 +172,6 @@
     </div>
   </div>
   {% endfor %}
-  <p style="margin-top: 50px"><em>These reports are generated from <a href="https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/tests/templates/test_report.html.j2">this code</a>, please reach out to the Connector Operations team for support.</em></p>
+  <p style="margin-top: 50px"><em>These reports are generated from <a href="https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/tests/templates/test_report.html.j2">this code</a>, please reach out to the Connector Operations team for support.</em></p>
```
Comment (pedroslopez) [inline airbyte-ci/connectors/pipelines/pipelines/tests/templates/test_report.html.j2:175]: This link was wrong after the move

File: airbyte-ci/connectors/pipelines/pipelines/bases.py [hunk 2]
```diff
@@ -620,3 +630,6 @@ def print(self):
 
         main_panel = Panel(Group(*to_render), title=main_panel_title, subtitle=duration_subtitle)
         console.print(main_panel)
+
+        if self.pipeline_context.dagger_cloud_url:
+            self.pipeline_context.logger.info(f"🔗 View commit in Dagger Cloud: {self.pipeline_context.dagger_cloud_url}")
```
Comment (pedroslopez) [inline airbyte-ci/connectors/pipelines/pipelines/bases.py:635]: Ideally i wanted to add this as a section in the Panel printed with console.print, but for the life of me I couldn't figure out how to make the link not get cut off. In CI it kept adding a newline which made the link not clickable.

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`02536c4f0d`](https://github.com/airbytehq/airbyte/pull/28947/commits/02536c4f0d8353d5e73577eb1411ea9ce0f5539d)) - ❌

⏲️  Total pipeline duration: 01mn54s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ❌        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci-internal/connectors/test/pull_request/pedroslopez_dagger-cloud-run-url/1690929216/02536c4f0d8353d5e73577eb1411ea9ce0f5539d/source-faker/4.0.0/output.html)

☁️ [View commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:02536c4f0d8353d5e73577eb1411ea9ce0f5539d)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`02536c4f0d`](https://github.com/airbytehq/airbyte/pull/28947/commits/02536c4f0d8353d5e73577eb1411ea9ce0f5539d)) - ❌

⏲️  Total pipeline duration: 02mn07s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ❌        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci-internal/connectors/test/pull_request/pedroslopez_dagger-cloud-run-url/1690929926/02536c4f0d8353d5e73577eb1411ea9ce0f5539d/source-faker/4.0.0/output.html)

☁️ [View commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:02536c4f0d8353d5e73577eb1411ea9ce0f5539d)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`e54b090911`](https://github.com/airbytehq/airbyte/pull/28947/commits/e54b0909115f3f9d00d4789324cb10ae11c8bbe6)) - ❌

⏲️  Total pipeline duration: 01mn54s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ❌        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci-internal/connectors/test/pull_request/pedroslopez_dagger-cloud-run-url/1690930340/e54b0909115f3f9d00d4789324cb10ae11c8bbe6/source-faker/4.0.0/output.html)

☁️ [View commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e54b0909115f3f9d00d4789324cb10ae11c8bbe6)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

File: airbyte-ci/connectors/pipelines/pipelines/dagger_run.py [hunk 3]
```diff
@@ -89,7 +89,7 @@ def check_dagger_cli_install() -> str:
 def main():
     os.environ[DAGGER_CLOUD_TOKEN_ENV_VAR_NAME_VALUE[0]] = DAGGER_CLOUD_TOKEN_ENV_VAR_NAME_VALUE[1]
     exit_code = 0
-    if sys.argv[1] == "--no-tui":
+    if len(sys.argv) > 1 and sys.argv[1] == "--no-tui":
```
Comment (pedroslopez) [inline airbyte-ci/connectors/pipelines/pipelines/dagger_run.py:92]: There was a bug with the latest changes here when no args were passed to airbyte-ci

File: airbyte-ci/connectors/pipelines/pipelines/bases.py [hunk 4]
```diff
@@ -618,5 +628,8 @@ def print(self):
         details_instructions = Text("ℹ️  You can find more details with step executions logs in the saved HTML report.")
         to_render = [step_results_table, details_instructions]
 
+        if self.pipeline_context.dagger_cloud_url:
+            self.pipeline_context.logger.info(f"🔗 View commit in Dagger Cloud: {self.pipeline_context.dagger_cloud_url}")
```
Comment (bnchrch) [inline airbyte-ci/connectors/pipelines/pipelines/bases.py:632]:
```suggestion
            self.pipeline_context.logger.info(f"🔗 For more information and better debugging view this run in Dagger Cloud: {self.pipeline_context.dagger_cloud_url}")
```

**Review by bnchrch — APPROVED** [review]
LGTM!

File: airbyte-ci/connectors/pipelines/pipelines/bases.py [hunk 5]
```diff
@@ -477,6 +478,9 @@ def print(self):
             failures_group = Group(*sub_panels)
             to_render.append(failures_group)
 
+        if self.pipeline_context.dagger_cloud_url:
+            self.pipeline_context.logger.info(f"🔗 View commit in Dagger Cloud: {self.pipeline_context.dagger_cloud_url}")
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/bases.py:482]:
```suggestion
            self.pipeline_context.logger.info(f"🔗 View runs for commit in Dagger Cloud: {self.pipeline_context.dagger_cloud_url}")
```

File: airbyte-ci/connectors/pipelines/pipelines/bases.py [hunk 6]
```diff
@@ -551,6 +556,10 @@ def post_comment_on_pr(self) -> None:
         ]
         markdown_comment += tabulate(report_data, headers=["Step", "Result"], tablefmt="pipe") + "\n\n"
         markdown_comment += f"🔗 [View the logs here]({self.html_report_url})\n\n"
+
+        if self.pipeline_context.dagger_cloud_url:
+            markdown_comment += f"☁️ [View commit in Dagger Cloud]({self.pipeline_context.dagger_cloud_url})\n\n"
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/bases.py:561]:
```suggestion
            markdown_comment += f"☁️ [View runs for commit in Dagger Cloud]({self.pipeline_context.dagger_cloud_url})\n\n"
```

File: airbyte-ci/connectors/pipelines/pipelines/bases.py [hunk 7]
```diff
@@ -618,5 +628,8 @@ def print(self):
         details_instructions = Text("ℹ️  You can find more details with step executions logs in the saved HTML report.")
         to_render = [step_results_table, details_instructions]
 
+        if self.pipeline_context.dagger_cloud_url:
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/bases.py:631]: Aren't you already logging this on line 481?

File: airbyte-ci/connectors/pipelines/pipelines/contexts.py [hunk 8]
```diff
@@ -172,6 +172,18 @@ def github_commit_status(self) -> dict:
     def should_send_slack_message(self) -> bool:
         return self.slack_webhook is not None and self.reporting_slack_channel is not None
 
+    @property
+    def is_dagger_cloud(self) -> bool:
+        return bool(os.getenv("_EXPERIMENTAL_DAGGER_CLOUD_TOKEN"))
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/contexts.py:177]:
```suggestion
        return "_EXPERIMENTAL_DAGGER_CLOUD_TOKEN" in os.environ
```

File: airbyte-ci/connectors/pipelines/pipelines/tests/templates/test_report.html.j2 [hunk 9]
```diff
@@ -128,6 +128,9 @@
     {% if dagger_logs_url %}
     <li><b><a href="{{ dagger_logs_url }}">Dagger logs</a></b></li>
     {% endif %}
+    {% if dagger_cloud_url %}
+    <li><b><a href="{{ dagger_cloud_url }}">Dagger Cloud</a></b></li>
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/tests/templates/test_report.html.j2:132]:
```suggestion
    <li><b><a href="{{ dagger_cloud_url }}">Dagger Cloud UI</a></b></li>
```

**Review by alafanechere — APPROVED** [review]
Nice! 
Only minor suggestions.

File: airbyte-ci/connectors/pipelines/pipelines/bases.py (hunk 7 above)
Comment (pedroslopez) [reply airbyte-ci/connectors/pipelines/pipelines/bases.py:631]: This is a different Report's `print`. One is ConnectorReport.print and the other is Report.print

File: airbyte-ci/connectors/pipelines/pipelines/bases.py (hunk 6 above)
Comment (pedroslopez) [reply airbyte-ci/connectors/pipelines/pipelines/bases.py]: done in [9ab0f68](https://github.com/airbytehq/airbyte/pull/28947/commits/9ab0f680bea0d851a62a4c2a0bfd530d222dd110)

File: airbyte-ci/connectors/pipelines/pipelines/bases.py (hunk 5 above)
Comment (pedroslopez) [reply airbyte-ci/connectors/pipelines/pipelines/bases.py]: done in [9ab0f68](https://github.com/airbytehq/airbyte/pull/28947/commits/9ab0f680bea0d851a62a4c2a0bfd530d222dd110)

File: airbyte-ci/connectors/pipelines/pipelines/contexts.py [hunk 10]
```diff
@@ -172,6 +172,18 @@ def github_commit_status(self) -> dict:
     def should_send_slack_message(self) -> bool:
         return self.slack_webhook is not None and self.reporting_slack_channel is not None
 
+    @property
+    def is_dagger_cloud(self) -> bool:
```
[the remark this answers is not in the document]
Comment (pedroslopez) [reply airbyte-ci/connectors/pipelines/pipelines/contexts.py]: done in [4854ee3](https://github.com/airbytehq/airbyte/pull/28947/commits/4854ee3dc7c9c2024d2995e39965ffa84a21f2c6)

**Pull request merged**